code/API_definitions/qod-provision.yaml

```diff
@@ -0,0 +1,876 @@
+
+    * **Identifier for the device**:
+    At least one identifier for the device (user equipment) out of four options: IPv4 address, IPv6 address, Phone number, or Network Access Identifier assigned by the network operator for the device, at the request time. When multiple identifiers are provided, they must identify the same device. After the provision request is accepted, the device may get different IP addresses, but the provision will still apply to the device that was identified during the request process.between the device and the network.
+
+    * **Notification URL and token**:
+    Developers may provide a callback URL on which notifications about all status change events (eg. provision termination) can be received from the service provider. This is an optional parameter.
+
+    # Resources and Operations overview
+    The API defines four operations:
+
+    - An operation to setup a new QoD provision for a given device.
+    - An operation to get the information about a specific QoD provision, identified by its `provisionId`.
+    - An operation to get the QoD provisions for a given device.
+    - An operation to terminate a QoD provision, identified by its `provisionId`.
+
+
+    ### Authorization and authentication
+
+    CAMARA guidelines defines a set of authorization flows which can grant API clients access to the API functionality, as outlined in the document [CAMARA-API-access-and-user-consent.md](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md). Which specific authorization flows are to be used will be determined during onboarding process, happening between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API, while also being subject to the prevailing legal framework dictated by local legislation.
+
+    It is important to remark that in cases where personal user data is processed by the API, and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of 3-legged access tokens becomes mandatory. This measure ensures that the API remains in strict compliance with user privacy preferences and regulatory obligations, upholding the principles of transparency and user-centric data control.
+
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: project-email@sample.com
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: wip
+externalDocs:
+  description: Product documentation at Camara
+  url: https://github.com/camaraproject/
+servers:
+  - url: "{apiRoot}/qod-provision/vwip"
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root, defined by the service provider, e.g. `api.example.com` or `api.example.com/somepath`
+tags:
+  - name: QoD Provision
+    description: Manage the provision of QoD
+paths:
+  /provisions:

```

Comment:
Is provisions the best name for this endpoint?  As I'm reviewing this, I keep thinking about getting supplies (provisions) for a camping trip.

What about `/persistent` or `/device-qod`?  Something that denotes it's setting a QoS Provide for the device and / or that it's a type of QoD session that will never expire.

Reply:
I trust a native speaker on this :) We are open to change the name to whatever makes the use case more understandable and remark the difference with respect to the other API (sessions). The concept is related to the [provisioning](https://en.wikipedia.org/wiki/Provisioning_(technology)) of the QoD persistently.

For API name, maybe `qod-provisioning` is more clear? For the resource name it could be `device-qos`, and it would imply operations such as  `POST /qod-provisioning/v0.1/device-qos` and `GET /qod-provisioning/v0.1/device-qos/{device_qos_id}`

Reply:
@jlurien /qod-provisioning makes sense to me.